scipy/signal/_upfirdn.py

```diff
@@ -72,14 +72,19 @@ def __init__(self, h, x_dtype, up, down):
             raise ValueError('Both up and down must be >= 1')
         # This both transposes, and "flips" each phase for filtering
         self._h_trans_flip = _pad_h(h, self._up)
+        self._h_trans_flip = np.ascontiguousarray(self._h_trans_flip)
 
-    def apply_filter(self, x):
-        """Apply the prepared filter to a 1D signal x"""
-        output_len = _output_len(len(self._h_trans_flip), len(x),
+    def apply_filter(self, x, axis=-1):
+        """Apply the prepared filter to the specified axis of a nD signal x"""
+        output_len = _output_len(len(self._h_trans_flip), x.shape[axis],
                                  self._up, self._down)
-        out = np.zeros(output_len, dtype=self._output_type)
-        _apply(np.asarray(x, self._output_type), self._h_trans_flip, out,
-               self._up, self._down)
+        output_shape = np.asarray(x.shape)
+        output_shape[axis] = output_len
+        out = np.zeros(output_shape, dtype=self._output_type, order='C')
+        axis = axis % x.ndim
+        _apply(np.ascontiguousarray(x, self._output_type),
```

Comment:
Replacing with ```np.asarray```? It seems ```_apply``` uses the strides which should not require x to be contiguous.